test_communication/CMakeLists.txt

```diff
@@ -162,6 +160,51 @@ if(BUILD_TESTING)
     endif()
   endmacro()
 
+  macro(custom_test_py)
+    # adding python tests
+    find_package(rmw_implementation_cmake REQUIRED)
+    set(TEST_PUBLISHER_PY_EXECUTABLE "${CMAKE_CURRENT_SOURCE_DIR}/test/publisher_py.py")
+    set(TEST_SUBSCRIBER_PY_EXECUTABLE "${CMAKE_CURRENT_SOURCE_DIR}/test/subscriber_py.py")
+    # TODO(mikaelarguedas) edit this once typesupport_introspection_c merged in fastrtps
+    # TODO(mikaelarguedas) edit this once connext_dynamic working with all message types
+    if(NOT  ("${rmw_implementation1}" STREQUAL "rmw_fastrtps_cpp" OR ${rmw_implementation1} STREQUAL "rmw_connext_dynamic_cpp"))
+      get_rmw_typesupport(typesupport_c_impl "${rmw_implementation1}" LANGUAGE "c")
+      set(TEST_RMW_IMPLEMENTATION_PY "${rmw_implementation1}")
+      if(NOT typesupport_c_impl STREQUAL " ")
+        foreach(message_file ${message_files})
+          get_filename_component(TEST_MESSAGE_TYPE "${message_file}" NAME_WE)
+          # TODO(dirk-thomas) publishing DynamicArrayPrimitives in OpenSplice is buggy
+          # https://github.com/ros2/rclcpp/issues/219
+          if(NOT "${TEST_MESSAGE_TYPE}" STREQUAL "DynamicArrayPrimitives" OR NOT "${rmw_implementation1}" STREQUAL "rmw_opensplice_cpp")
```

Comment:
The string compare doesn't use a trailing space. So for "older" CMake versions that might be a problem. Since we move to CMake 3.5 soon it might not be worth fixing.